mindmeld/components/entity_resolver.py

```diff
@@ -562,38 +776,829 @@ def _predict_exact_match(self, entity):
 
         values = []
         for cname in cnames:
-            for item in self._exact_match_mapping["items"][cname]:
+            for item in self._entity_mapping["items"][cname]:
                 item_value = copy.copy(item)
                 item_value.pop("whitelist", None)
                 values.append(item_value)
 
         return values
 
-    def load(self):
-        """Loads the trained entity resolution model from disk."""
-        try:
-            if self._use_text_rel:
-                scoped_index_name = get_scoped_index_name(
-                    self._app_namespace, self._es_index_name
+    def _load(self):
+        self.fit()
+
+
+class SentenceBertCosSimEntityResolver(EntityResolverModelBase):
+    """
+    Resolver class for bert models as described here:
+    https://github.com/UKPLab/sentence-transformers
+    """
+
+    TRANSFORMER_MODEL, POOLING_MODEL, SBERT_MODEL = None, None, None
+
+    def __init__(self, app_path, resource_loader, entity_type, er_config, **_kwargs):
+        super().__init__(app_path, resource_loader, entity_type, er_config)
+        self._entity_mapping = None
+        self._synonyms = None
+        self._synonyms_embs = None
+        self._use_sbert_model = False
+        self._model_pretrained_name_or_abspath = (
+            self.er_config["model_settings"]["pretrained_name_or_abspath"]
+        )
+        self.quantize_model = self.er_config["model_settings"]["quantize_model"]
+
+        # init BERT model
+        if not SentenceBertCosSimEntityResolver.TRANSFORMER_MODEL:
```

Comment:
Its unclear why we need this conditional when `TRANSFORMER_MODEL ` always seems to be None.

Reply:
because they are class vars, they are None only the first time a SentenceBertCosSimEntityResolver is initialized. For any subsequent initializations, they are not None, right?